src/Controls/src/Core/SearchBar/SearchBar.cs

```diff
@@ -115,6 +116,28 @@ public SearchBar()
 			_platformConfigurationRegistry = new Lazy<PlatformConfigurationRegistry<SearchBar>>(() => new PlatformConfigurationRegistry<SearchBar>(this));
 		}
 
+		private protected override void OnHandlerChangingCore(HandlerChangingEventArgs args)
+		{
+			base.OnHandlerChangingCore(args);
+
+			if (Application.Current == null)
+				return;
+
+			if (args.NewHandler == null)
```

Comment:
You're only unsubscribing when `NewHandler` is null, but you should unsubscribe whenever there's an `OldHandler`.